src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -173,175 +174,179 @@ protected void startLoadBalancer() {
-            api.getLoadBalancerApi().createListeningInSubnetsAssignedToSecurityGroups(elbName, subnets, securityGroups != null ? securityGroups : ImmutableList.<String>of());
-
-            // Reset the health check
-            /*
-            HealthCheck oldHealthCheck = loadBalancer.getHealthCheck();
-            if (healthCheckEnabled != null && healthCheckEnabled) {
-                // TODO remove duplication from createLoadBalancer; would require passing in a huge amount of config
-                // as parameters, or the shared method reading the config itself
-                String targetTemplate = getConfig(HEALTH_CHECK_TARGET);
-                Map<String, Object> substitutions = ImmutableMap.<String, Object>builder()
-                        .put("instancePort", instancePort)
-                        .put("instanceProtocol", instanceProtocol)
-                        .build();
-                String target = TemplateProcessor.processTemplateContents(targetTemplate, substitutions);
-
-                HealthCheck healthCheck = HealthCheck.builder()
-                        .target(target)
-                        .interval(healthCheckInterval)
-                        .timeout(healthCheckTimeout)
-                        .healthyThreshold(healthCheckHealthyThreshold)
-                        .unhealthyThreshold(healthCheckUnhealthyThreshold)
-                        .build();
-
-                api.getHealthCheckApi().configureHealthCheck(healthCheck, elbName);
-            } else {
-                // TODO Not removing old health check because passing in null for ConfigureHealthCheckRequest gives:
-                //      Caused by: AmazonServiceException: Status Code: 400, AWS Service: AmazonElasticLoadBalancing, AWS Request ID: 47a2b87d-30ef-11e3-bd82-6571b2f68bd5, AWS Error Code: ValidationError, AWS Error Message: 1 validation error detected: Value null at 'healthCheck' failed to satisfy constraint: Member must not be null
-                if (oldHealthCheck != null) {
-                    LOG.warn("Existing ELB {} (in {}) health check ({}) not removed (no new health check wanted; continuing",
-                            new Object[] {elbName, this, oldHealthCheck});
-                }
-            }
-            */
 
-            sensors().set(Attributes.HOSTNAME, loadBalancer.getDnsName());
-            } finally {
-            Closeables2.closeQuietly(loadBalancerServiceContext.unwrap());
+        }
+    }
+
+    private void configureListeners(String elbName, LoadBalancer loadBalancer) {
+        Set<ListenerWithPolicies> listenerWithPolicies = loadBalancer.getListeners();
+        Set<Integer> listenerPorts = Sets.newHashSet();
+        for (ListenerWithPolicies listenerWithPolicy : listenerWithPolicies) {
+            listenerPorts.add(listenerWithPolicy.getPort());
         }
 
+        if (!listenerPorts.isEmpty()) {
+            getELBApi().getLoadBalancerApi().deleteLoadBalancerListeners(elbName, listenerPorts);
```

Comment:
Don't see `deleteLoadBalancerListeners` on `LoadBalancerApi` - probably to be PR'ed separately?